diff -ruN linux-vanilla-4.9.86/arch/powerpc/sysdev/ppc4xx_ocm.c linux-4.9.86/arch/powerpc/sysdev/ppc4xx_ocm.c
--- linux-vanilla-4.9.86/arch/powerpc/sysdev/ppc4xx_ocm.c	2018-03-03 03:23:29.000000000 -0600
+++ linux-4.9.86/arch/powerpc/sysdev/ppc4xx_ocm.c	2018-03-05 10:09:26.048895027 -0600
@@ -109,7 +109,9 @@
 	struct ocm_info *ocm;
 
 	const unsigned int *cell_index;
+	const char *status;
 	const unsigned int *cache_size;
+
 	int len;
 
 	struct resource rsrc;
@@ -124,7 +126,10 @@
 	}
 	ocm->index = *cell_index;
 
-	if (of_device_is_available(node))
+	status = of_get_property(node, "status", &len);
+	if (status && strcmp(status,"disabled") == 0)
+		ocm->status = OCM_DISABLED;
+	else
 		ocm->status = OCM_ENABLED;
 
 	cache_size = of_get_property(node, "cached-region-size", &len);
